circuits/circom/processMessages.circom

```diff
@@ -295,8 +295,7 @@ template ProcessMessages(
             }
         }
 
-        stateRootQip.leaf <== stateRootHasher.hash;
-        stateLeafQip.root === stateRoots[i + 1];
```

Comment:
I believe we want this check to make sure the state root proofs include the next state root to be processed

Reply:
the stateLeafQip is not defined and only used here, so I removed it. Is it needed here? stateRootQip.leaf changed to stateRootQip[i].leaf.

Reply:
Yes it should be set in the top level of the loop